src/scripts/OutSystems/OSUI/Patterns/SearchAPI.ts

```diff
@@ -0,0 +1,165 @@
+		}
+
+		const _newSearch = new OSFramework.OSUI.Patterns.Search.Search(searchId, JSON.parse(configs));
+		_searchMap.set(searchId, _newSearch);
+		return _newSearch;
+	}
+
+	/**
+	 * Function that will destroy the instance of the given Search
+	 *
+	 * @export
+	 * @param {string} searchId
+	 */
+	export function Dispose(searchId: string): string {
+		const result = OutSystems.OSUI.Utils.CreateApiResponse({
+			errorCode: ErrorCodes.Search.FailDispose,
+			callback: () => {
+				const search = GetSearchById(searchId);
+
+				search.dispose();
+
+				_searchMap.delete(searchId);
+			},
+		});
+
+		return result;
+	}
+
+	/**
+	 * Fucntion that will return the Map with all the Search instances at the page
+	 *
+	 * @export
+	 * @return {*}  {Array<string>}
+	 */
+	export function GetAllSearches(): Array<string> {
+		return OSFramework.OSUI.Helper.MapOperation.ExportKeys(_searchMap);
+	}
+
+	/**
+	 * Function that gets the instance of Search, by a given ID.
+	 *
+	 * @export
+	 * @param {string} searchId
+	 * @return {*}  {OSFramework.OSUI.Patterns.Search.ISearch}
+	 */
+	export function GetSearchById(searchId: string): OSFramework.OSUI.Patterns.Search.ISearch {
+		return OSFramework.OSUI.Helper.MapOperation.FindInMap(
+			'Search',
+			searchId,
+			_searchMap
+		) as OSFramework.OSUI.Patterns.Search.ISearch;
+	}
+
+	/**
+	 * Function that will initialize the pattern instance.
+	 *
+	 * @export
+	 * @param {string} searchId
+	 * @return {*}  {OSFramework.OSUI.Patterns.Search.ISearch}
+	 */
+	export function Initialize(searchId: string): OSFramework.OSUI.Patterns.Search.ISearch {
+		const search = GetSearchById(searchId);
+
+		search.build();
+
+		return search;
+	}
+
+	/**
+	 * Function to register a provider callback
+	 *
+	 * @export
+	 * @param {string} searchId
+	 * @param {string} eventName
+	 * @param {OSFramework.OSUI.GlobalCallbacks.OSGeneric} callback
+	 * @return {*} {string} Return Message Success or message of error info if it's the case.
+	 */
+	export function RegisterCallback(
+		searchId: string,
+		eventName: string,
+		callback: OSFramework.OSUI.GlobalCallbacks.OSGeneric
+	): string {
+		const result = OutSystems.OSUI.Utils.CreateApiResponse({
+			errorCode: ErrorCodes.Search.FailRegisterCallback,
+			callback: () => {
+				const _SearchItem = this.GetSearchById(searchId);
+
+				_SearchItem.registerCallback(eventName, callback);
+			},
+		});
+
+		return result;
+	}
+
+	/**
+	 * Function that will enable the native behavior
+	 *
+	 * @export
+	 * @param {string} searchId
+	 * @return {*}  {string}
+	 */
+	export function EnableNativeBehavior(searchId: string): string {
+		const result = OutSystems.OSUI.Utils.CreateApiResponse({
+			errorCode: ErrorCodes.Search.FailEnableNativeBehavior,
+			callback: () => {
+				const search = this.GetSearchById(searchId);
+
+				search.enableNativeBehavior();
+			},
+		});
+
+		return result;
+	}
+
+	export function UpdateGlassButtonAriaLabel(searchId: string, ariaLabel: string): string {
```

Comment:
Missing a comment explaining the function as we have for the others

Reply:
Done